### PR TITLE
Add posts lists

### PR DESCRIPTION
**Pull request by mfpiccolo**
This PR adds a the posts list container component, route, actions, constants, reducer and store.

### CODE REVIEW

File: src/reducers/posts.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import _ from 'lodash';
+import {PostsTypes as types} from 'constants';
+
+const initState = {
+    status: 'not fetched',
+    error: null,
+    data: []
+};
+
+export default function reducePosts(state = initState, action) {
```
Comment (hmhealey) [inline src/reducers/posts.js]: Nothing to change here yet, but we're going to need to reorganize the store still to fit what we discussed last week. Ideally, the response to this call should add these posts to a map in `store.entities.posts.posts` and then their ids into `store.entities.posts.postIdsByChannelIds`. The status/error state will also need to be moved out into `store.requests`.

File: src/components/channels_list.js [hunk 2]
```diff
@@ -40,8 +40,7 @@ class ChannelsList extends Component {
     componentWillReceiveProps(props) {
         if (props.channels.current_channel_id &&
           !this.props.channels.current_channel_id) {
-            console.warn('TODO route to PostsList'); // eslint-disable-line no-console
-            // TODO: Routes.PostLists(); // eslint-disable-line new-cap
+            Routes.PostsList(); // eslint-disable-line new-cap
```
Comment (hmhealey) [inline src/components/channels_list.js]: Why is this method name capitalized?

Comment (mfpiccolo) [inline src/components/channels_list.js]: This is just part of the react-native-router-flux api.  It comes from the routes.js file.

```
                    <Scene
                        key='PostsList'
                        component={PostsList}
                        title='Posts List'
                    />
```

We can change the key to be camel case.  Perhaps `goToPostsList`

Comment (thomchop) [inline src/components/channels_list.js]: I would rather create a wrapper around the router/routes/route-actions so we can call `Router.go('PostsList');`. By naming our routes like functions I think we'll end up with confusing name overloading.
e.g. `Actions.selectTeam();` and `props.actions.selectTeam();`.

File: src/components/posts_list.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        flexDirection: 'column',
+        alignItems: 'center',
+        paddingTop: 200,
+        backgroundColor: 'white'
+    }
+});
+
+const propTypes = {
+    posts: PropTypes.object.isRequired,
+    current_team_id: PropTypes.string.isRequired,
```
Comment (thomchop) [inline src/components/posts_list.js]: I have a nuanced preference here: camel case for function/method names and snake case for other variables/properties. I've realized that's too nuanced for most people to buy into. As Mike said, I'm going against the grain. So I'm learning to love camel case everywhere on this project.
However 4-space indentation is another matter, but I'll keep quiet about that...

**Comment by mfpiccolo** [comment]
@hmhealey Made the changes

File: src/client/client.js [hunk 4]
```diff
@@ -249,6 +249,15 @@ export default class Client {
     }
 
     // Post routes
+    fetchPosts = (channelId, onRequest, onSuccess, onFailure) => {
+        return this.doFetch(
+            `${this.getPostsRoute(channelId)}/page/0/60`,
```
Comment (thomchop) [inline src/client/client.js]: When we implement pagination of posts we'll pass in the `page` and `offset` params so this function is more like `Client.getPostsPage` in the webapp.

Comment (mfpiccolo) [inline src/client/client.js]: 👍

**Comment by hmhealey** [comment]
Works for me. Thanks!

**Pull request merged**